azurerm/internal/services/policy/data_source_policy_definition.go

```diff
@@ -68,39 +72,24 @@ func dataSourceArmPolicyDefinitionRead(d *schema.ResourceData, meta interface{})
 	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	name := d.Get("display_name").(string)
+	displayName := d.Get("display_name").(string)
+	name := d.Get("name").(string)
 	managementGroupID := d.Get("management_group_id").(string)
 
-	var policyDefinitions policy.DefinitionListResultIterator
-	var err error
-
-	if managementGroupID != "" {
-		policyDefinitions, err = client.ListByManagementGroupComplete(ctx, managementGroupID)
-	} else {
-		policyDefinitions, err = client.ListComplete(ctx)
-	}
-
-	if err != nil {
-		return fmt.Errorf("Error loading Policy Definition List: %+v", err)
-	}
-
 	var policyDefinition policy.Definition
-
-	for policyDefinitions.NotDone() {
-		def := policyDefinitions.Value()
-		if def.DisplayName != nil && *def.DisplayName == name {
-			policyDefinition = def
-			break
+	var err error
+	if displayName != "" {
+		policyDefinition, err = getPolicyDefinitionByDisplayName(ctx, client, displayName, managementGroupID)
+		if err != nil {
+			return fmt.Errorf("failed to read Policy Definition (Display Name %q): %+v", displayName, err)
 		}
-
-		err = policyDefinitions.NextWithContext(ctx)
+	} else if name != "" {
+		policyDefinition, err = getPolicyDefinition(ctx, client, name, managementGroupID)
```

Comment:
could we rename this to
```suggestion
		policyDefinition, err = getPolicyDefinitionByName(ctx, client, name, managementGroupID)
```

and place in `policy.go` for all policy helpers?

Reply:
Done